modules/postgresql/main.tf

```diff
@@ -173,6 +174,33 @@ resource "google_sql_user" "additional_users" {
   depends_on = [null_resource.module_depends_on, google_sql_database_instance.default]
 }
 
+resource "google_sql_user" "iam_account" {
+  for_each = local.iam_users
+  project  = var.project_id
+  name = each.value.is_account_sa ? (
+    trimsuffix(each.value.email, ".gserviceaccount.com")
+    ) : (
+    each.value.email
+  )
+  instance = google_sql_database_instance.default.name
+  type     = each.value.is_account_sa ? "CLOUD_IAM_SERVICE_ACCOUNT" : "CLOUD_IAM_USER"
+  depends_on = [
+    null_resource.module_depends_on,
+    google_project_iam_member.iam_binding,
+  ]
+}
+
+resource "google_project_iam_member" "iam_binding" {
+  for_each = local.iam_users
+  project  = var.project_id
+  role     = "roles/cloudsql.instanceUser"
+  member = each.value.is_account_sa ? (
+    "serviceAccount:${each.value.email}"
+    ) : (
+    "user:${each.value.email}"
+  )
+}
```

Comment:
nit: suggest we move this up a block for readability as the previous block depends on this `google_project_iam_member.iam_binding,`

Reply:
sure

Reply:
done